### PR TITLE
feat: Add initial tvOS support to some firebase packages

### PR DESCRIPTION
**Pull request by adamkoch**
Adds compatibility with tvOS (using [react-native-tvos](https://github.com/react-native-tvos/react-native-tvos)) for RNFirebase packages: app, auth, firestore, functions, storage and app check

### Description

Two main changes:

 - Updates the podspec files of packages to add tvOS deployment targets.
 - Updates `RNFBAuthModule.m` to only remove incompatible tvOS APIs

This allows running the updated Firebase packages on tvOS.

### Related issues

Not aware of an open issue but discussion here:
https://github.com/invertase/react-native-firebase/discussions/5722#discussioncomment-11125191

### Release Summary

Adds compatibility with tvOS (using [react-native-tvos](https://github.com/react-native-tvos/react-native-tvos)) for RNFirebase packages: app, auth, firestore, functions, storage and app check

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [X] Yes
- My change supports the following platforms;
  - [ ] `Android`
  - [X] `iOS` (well sort of, `tvOS`)
- My change includes tests;
  - [ ] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [ ] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [X] No

### Test Plan

Not sure how to add a test plan for this change.

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: packages/app-check/RNFBAppCheck.podspec [hunk 1]
```diff
@@ -10,6 +10,7 @@ if coreVersionDetected != coreVersionRequired
 end
 firebase_ios_target = appPackage['sdkVersions']['ios']['iosTarget']
 firebase_macos_target = appPackage['sdkVersions']['ios']['macosTarget']
+firebase_tvos_target = appPackage['sdkVersions']['ios']['tvosTarget']
```
Comment (douglowder) [inline packages/app-check/RNFBAppCheck.podspec:13]: You can simplify this and just keep tvOS target the same as iOS.

File: packages/app/package.json [hunk 2]
```diff
@@ -76,7 +76,8 @@
     "ios": {
       "firebase": "11.4.0",
       "iosTarget": "13.0",
-      "macosTarget": "10.15"
+      "macosTarget": "10.15",
+      "tvosTarget": "12.0"
```
Comment (douglowder) [inline packages/app/package.json:80]: This should be 13.4 or higher for compatibility with RNTV 0.73 and later

Comment (adamkoch) [reply packages/app/package.json:80]: Bit of a catch :) If we use firebase_ios_target for tvOS then there isn't a way set tvosTarget to to 13.4 without also setting iosTarget, which I'm not sure we want to do when most users aren't targeting tvOS?

Comment (mikehardy) [reply packages/app/package.json:80]: I'm okay with having these separate, and we align the minimums with the firebase-ios-sdk podspecs not with react-native. The Auth podspec over there is the highest one so it determines all, and it is currently at 13.0:

https://github.com/firebase/firebase-ios-sdk/blob/ae2554220afb03a5d6fc389c28a86061a12c7caa/FirebaseAuth.podspec#L22-L25

So in final form what I will do here is keep them separate, but bump to 13.0

**Comment by mikehardy** [comment]
I couldn't push to this fork:branch combo with my fixups so I've posted them all here:

- #8111 

I will close this PR but obviously it is the base of that one - if you could head over there and check the add-on commits to see the changes I made, I think they should work?

You can grab the patch set for testing via patch-package by following the details link on the CI patch-package workflow and looking at the summary there

**Pull request closed without merging**